torch/_dynamo/guards.py

```diff
@@ -159,10 +159,11 @@ def set_export_info(self, guard_type, guarded_class, code_list, obj_weakref):
         ), "Guarded class id must be identical, or None"
         self.guarded_class_weakref = guarded_class
 
-        if not self.code_list:
-            self.code_list = code_list
-        else:
-            self.code_list.extend(code_list)
+        if code_list:
+            if not self.code_list:
+                self.code_list = code_list
+            else:
+                self.code_list.extend(code_list)
```

Comment:
cursed conditional lol

Reply:
It's alright, tensor_ref just doesn't get a code list.